bin/cloudgov/audit/cf-components

```diff
@@ -0,0 +1,270 @@
+    OPTARG="${OPTARG#=}"      # if long option argument, remove assigning `=`
+  fi
+  case "$OPT" in
+    a | all )      all=true ;;
+    s | spaces )   needs_arg; spaces="$OPTARG" ;;
+    \? )           exit 2 ;;  # bad short option (error reported via getopts)
+    * )            die "Illegal option --$OPT" ;;  # bad long option
+  esac
+done
+shift $((OPTIND-1)) # remove parsed options and args from $@ list
+
+getPaginationCount() { # $1 = resource, $2 = relationship, $3 = relationship id
+  PAGES=$(cf curl "/v3/$1" | jq -r '.pagination | .total_pages');
+  SELECT=""
+  if [ -n "$2" ]; then
+    SELECT=' select( .relationships.'"$2"'.data.guid == "'"$3"'" ) |'
+  fi
+  for i in $(seq 1 "$PAGES"); do
+    IDS=${IDS}$(cf curl "/v3/$1?page=$i&per_page=50" | jq -r '[.resources[] |'"$SELECT"' .guid]')
+  done
+  echo "$IDS"
+}
+
+echo "${bold}${underline}CF Components${nounderline}${normal}"
+
+# Orgs->Domains/Spaces->Apps/Services->Tasks/Sidecars
+if [ $all == true ]; then
+  ORGIDS=$(getPaginationCount "organizations")
+else
+  echo "Showing info for gsa-tts-usagov, add --all argument to see all orgs."
+  ORGIDS=$(cf curl "/v3/organizations" | jq -r '[.resources[] | select(.name == "gsa-tts-usagov") | .guid]')
+fi
+
+IFS=',' read -r -a spaces_array <<< "$spaces"
+
+echo "$ORGIDS" | jq -r '.[]' | while read -r ORGID; do
+  ORG=$(cf curl "/v3/organizations/$ORGID")
+  ORGNAME=$(echo "$ORG" | jq -r '.name')
+  ORGCREATE=$(echo "$ORG" | jq -r '.created_at')
+  ORGUPDATE=$(echo "$ORG" | jq -r '.updated_at')
+
+  echo "Org: ${underline}$ORGNAME${nounderline} (Last updated: $ORGUPDATE | Created: $ORGCREATE)"
+
+  DOMIANIDS=$(getPaginationCount "domains" "organization" "$ORGID")
+  if [ "$DOMIANIDS" != "[]" ]; then
+    echo "$level2 Domains:"
+    echo "$DOMIANIDS" | jq -r '.[]' | while read -r DOMIANID; do
+      DOMAIN=$(cf curl "/v3/domains/$DOMIANID")
+      DOMAINNAME=$(echo "$DOMAIN" | jq -r '.name')
+      DOMAINCREATED=$(echo "$DOMAIN" | jq -r '.created_at')
+      DOMAINUPDATED=$(echo "$DOMAIN" | jq -r '.updated_at')
+      DOMAININTERNAL=$(echo "$DOMAIN" | jq -r '.internal')
+      DOMAINPROTOCOLS=$(echo "$DOMAIN" | jq -c '.supported_protocols')
+      echo "$level3 $DOMAINNAME (Last updated: $DOMAINUPDATED | Created: $DOMAINCREATED | Internal: $DOMAININTERNAL | Supported Protocols: $DOMAINPROTOCOLS)"
+    done
+  fi
+
+  if [ $spaces == "all" ]; then
+    SPACEPAGES=$(cf curl "/v3/spaces" | jq -r '.pagination | .total_pages');
+    if [[ $SPACEPAGES -gt 1 ]]; then
+      for i in $SPACEPAGES; do
+        SPACEIDS=$(cf curl "/v3/spaces?page=$i&per_page=50" | jq -r '.resources[] | select( .relationships.organization.data.guid == "'$ORGID'" ) | .guid')
+      done
+    else
+      SPACEIDS=$(cf curl "/v3/spaces" | jq -r '.resources[] | select( .relationships.organization.data.guid == "'$ORGID'" ) | .guid')
+    fi
+  else
+    SPACEPAGES=cf curl "/v3/spaces" | jq -r '.pagination | .total_pages';
```

Comment:
Another error here `curl: (3) URL rejected: No host part in the URL` -- I ran the script with a list of spaces specified. 
(It didn't actually affect my results because there weren't enough spaces to require pagination.)